### DISCUSSION TITLE
Why Is The Variable Font A .woff2 File?

### DISCUSSION THREAD

**Discussion by Bronnel721** [ivodolenc/aspekta#7] [Q&A] [answered]
It should be a .ttf file, not a .woff2 file.

**Comment by ivodolenc** [comment]
It's assumed that the `Variable Font` is used mostly on the web. `Variable Font` contains some overlaps and this is something you want to avoid if you use font in Desktop environment.

For example, if you have some text that needs to be expanded to vectors, let's say in some printable graphics, then you would have to deal with all overlaps that preserved in the font which may be overkill for some printers or cuters.

This is one of the disadvantages of `VF` for desktop.

I suggest `otf` or `ttf` for desktop use, and you also get extra weights `50`, `950` and `1000`.

But if you need `VF.ttf` format it can be super-easily generated from the source with a few clicks.

**Comment by Bronnel721** [comment]
I use FontForge and it’s hard to make variable fonts with it.

**Comment by ivodolenc** [comment]
Ah ok, can you tell some advantages of using variable font in `ttf` format or if you have any special case to mention?

I can export it in the next release in `ttf` and `woff2`.

**Comment by ivodolenc** [comment]
This is also a good topic for discussion and exchange of ideas and advice.

I just test exported `VF` (v1.300) and here are some info:

### Web

- Collection / 100-900 weights
- Contains overlaps (bad for desktop usage)
- 1 file `VF.ttf` ~31KB
- 1 file `VF.woff2` ~17KB

### Desktop

- Full Collection / 50-1000 weights
- No overlaps
- 20 files `otf` ~353KB
- 20 files `ttf` ~590KB

**Comment by Bronnel721** [comment]
If I want to use it on Inkscape, that would be impossible. Because I can’t install .woff2 files in Windows 11. And how come every type foundry does the exact opposite of what you do like [Omnibus](https://github.com/Omnibus-Type/Asap/tree/master/fonts/variable)-[Type](https://github.com/Omnibus-Type/Asap/tree/master/fonts/webfonts), my favorite font foundry?

**Comment by ivodolenc** [comment, marked as answer]
Each project is different and specific in its own way, which is why certain decisions were made.

I don't want to encourage users to do something that they might have to deal with later and thus take extra time for some ridiculous mistakes that they could have avoided at the start.

This is the reason why I did not export the Variable Font in .ttf format, because as I mentioned before, static .otf and .ttf formats are intended for the Desktop environment.

If you can install VF in .ttf format then you can also install both .otf and .ttf formats on any system.

In the [Roadmap](https://github.com/users/ivodolenc/projects/2), I stated right at the beginning that the variable font should be improved in the sense that all masters should be adjusted specifically due to the paths overlapping in some glyphs.

**Why are there overlaps in the source file?**

- It gives additional flexibility in creating various shapes and components which ensures the consistency of the entire font.

I want to keep overlaps in the main masters because it helps a lot in the development and maintenance process. Currently the focus is to add support for most languages ​​and after that I will release a new Major update (v2.000) which will have an additional `AspektaVF` source file, tailored specifically for VF without path overlaps.

**Aspekta** is an open source project and publicly available to everyone so you can always export all the formats you need for your project.

Feel free to explore additional information about Variable Fonts and their uses, as well as their pros and cons.

**Comment by ivodolenc** [comment]
**Aspekta Variable** is now available in `.ttf` and `.woff2` formats. [More info](https://github.com/ivodolenc/aspekta/releases)
